tests/xharness/AppRunner.cs

```diff
@@ -185,21 +186,20 @@ public async Task<ProcessExecutionResult> InstallAsync (CancellationToken cancel
 
 			FindDevice ();
 
-			var args = new List<string> ();
+			var args = new MlaunchArguments ();
+
 			if (!string.IsNullOrEmpty (harness.XcodeRoot)) {
-				args.Add ("--sdkroot");
-				args.Add (harness.XcodeRoot);
+				args.Add (new SdkRootArgument (harness.XcodeRoot));
 			}
+
 			for (int i = -1; i < harness.Verbosity; i++)
-				args.Add ("-v");
-			
-			args.Add ("--installdev");
-			args.Add (AppInformation.AppPath);
-			AddDeviceName (args, companionDeviceName ?? deviceName);
+				args.Add (new VerbosityArgument ());
+
+			args.Add (new InstallAppOnDeviceArgument (AppInformation.AppPath));
+			args.Add (new DeviceNameArgument (companionDeviceName ?? deviceName));
 
 			if (runMode == RunMode.WatchOS) {
-				args.Add ("--device");
-				args.Add ("ios,watchos");
+				args.Add (new DeviceArgument("ios,watchos"));
```

Comment:
minor: missing space before `(`

Reply:
Would it make more sense to accept `params` here ?
i.e. `new DeviceArgument ("ios", "watchos");`

Reply:
Looks like these two are a value of itself telling `mlaunch` to do two companion device